content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,314 @@
+
+## FBA and Separating Sybil Attack Resistance from Consensus by Making it a Local Selection
+
+### Consensus
+
+A consensus algorithm is an agreement system that follows a predefined set of steps in order to allow a group of participants (nodes) to reach to the same decision about a statement.
+In a blockchain network nodes repeatedly decide what the complete history of a shared ledger is, from among multiple possible versions that occasionally conflict between each other.
+This network-wide agreement allows the recipient of a crypto coin to have faith that the coin is both (a) valid (not counterfeit) and (b) not already spent elsewhere.
+
+### Federated Byzantine Agreement
+
+#### Introduction
+
+Any agreement system in a distributed computing network needs to be fault-tolerant.
+This means it must produce consistent results despite errors like slow communication links, unresponsive nodes and misordered messages.
+A Byzantine agreement system is additionally tolerant of [`Byzantine`](https://medium.com/coinmonks/a-note-from-anthony-if-you-havent-already-please-read-the-article-gaining-clarity-on-key-787989107969) faults: nodes that give false information, whether due to error or in a deliberate attempt to subvert the system or gain some advantage.
+Such a system prevents the double spending problem by using a form of majority called `quorum`.
+A node in such a network refuses to commit to a particular version of the ledger history until it sees that enough of its peers are also prepared to commit to it.
+The number of peers that is enough to make a decision is the quorum and once each of the peers commit to a value after its quorum had committed, the system form a large enough voting bloc to force the remaining nodes to agree with the decision.
+In such a system malicious nodes attempts to create double spending events, or in other way invalid transactions, will be overwhelmed by the network votes from honest nodes.
+
+In order a majority to be formed, the number of all nodes in the system must be known and its quorum to be calculated at any time.
+Opposed to this concept, in a Federated Byzantyne Agreement System (FBAS) the participants in the network are loosely defined and can join and leave at any time without coordinating with any central authority.
+In such a system quorums are incomplete membership snapshots that can change at any time.
+This makes the system consensus probabilistic with some error size (probability of not reaching a consensus) present at all time.
+
+The network does not trust a single entity to operate the network, but rather works as a decentralized system, in which each participant knows only about a group of others peers, but not about all the nodes in the network.
+In the FBA system each node has knowledge of other nodes that it considers important.
```

Comment:
This is corrected https://github.com/optakt/optakt.github.io/pull/21/commits/f524c8d94d32f6edd6da60a9e7f74a848c44d4ce#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R31
But does it matter?

Reply:
It's just that since it's referred to as FBAS in other places, I think it's better to be consistent and call it FBAS everywhere or FBA System everywhere. Either way is fine for me, but consistency is key in technical writing :)